File: src/RuleEngine/RuleContext.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace RuleEngine
{
    public class RuleContext
    {
        public RuleContext(string name)
        {
            Name = name;
        }

        public Guid Id { get; set; } = Guid.NewGuid();
        public string Name { get; set; }
        public List<IRuleElement> Elements { get; set; } = new List<IRuleElement>();

        public RuleContext AddProposition(string statement, bool value)
        {
            Elements.Add(new Proposition(statement, value));

            return this;
        }

        public RuleContext AddVariable(string name, object value)
        {
            Elements.Add(new Variable(name, value));

            return this;
        }

        public IRuleElement Find(string name)
        {
            int count = Elements.Count;
            for (int i = 0; i < count; i++)
            {
                var e = Elements[i];
                if (e.Name == name)
                    return e;
            }

            return default(IRuleElement);
        }

        public RuleContext Append(RuleContext context)
        {
            foreach (var e in context.Elements)
            {
                Elements.Add(e);
            }

            return this;
        }
    }
}

File: src/RulesEngine.WebApp/Scripts/rules-engine/RuleContext.ts
import { IRuleElement } from "./IRuleElement"
import { Proposition } from "./Proposition"
import { Variable } from "./Variable"

export class RuleContext {
    constructor(name: string) {
        this.Name = name;
    }

    public Id: string = "";
    public Name: string = "";
    public Elements: Array<IRuleElement> = [];

    public AddProposition(statement: string, value: boolean): RuleContext {
        this.Elements.push(new Proposition(statement, value));
        return this;
    }

    public AddVariable(name: string, value: any): RuleContext {
        this.Elements.push(new Variable(name, value));
        return this;
    }

    public Find(name: string): IRuleElement {
        let count: number = this.Elements.length;
        let e: IRuleElement;
        for (let i: number = 0; i < count; i++){
            e = this.Elements[i];
            if (e.Name === name)
                break;
            e = null;
        }

        return e;
    }

    public Append(context: RuleContext): RuleContext {
        let count: number = context.Elements.length;
        for (let i: number = 0; i < count; i++) {
            let e: IRuleElement = context.Elements[i];
            this.Elements.push(e);
        }

        return this;
    }
}